redfish-core/lib/assembly.hpp

```diff
@@ -407,15 +560,83 @@ inline void setAssemblyLocationIndicators(
         {
             setLocationIndicatorActive(asyncResp, assembly, iter->second);
         }
+
+        auto iter2 = oemIndicatorMap.find(std::to_string(assemblyIndex));
+
+        if (iter2 != oemIndicatorMap.end())
+        {
+            std::optional<nlohmann::json> openbmc;
+            if (!json_util::readJson(iter2->second, asyncResp->res, "OpenBMC",
+                                     openbmc))
+            {
+                BMCWEB_LOG_ERROR("Property Value Format Error ");
+                messages::propertyValueFormatError(
+                    asyncResp->res,
+                    (*openbmc).dump(2, ' ', true,
+                                    nlohmann::json::error_handler_t::replace),
+                    "OpenBMC");
+                return;
+            }
+
+            if (!openbmc)
+            {
+                BMCWEB_LOG_ERROR("Property Missing ");
+                messages::propertyMissing(asyncResp->res, "OpenBMC");
+                return;
+            }
+
+            std::optional<bool> readytoremove;
+            if (!json_util::readJson(*openbmc, asyncResp->res, "ReadyToRemove",
+                                     readytoremove))
+            {
+                BMCWEB_LOG_ERROR("Property Value Format Error ");
+                messages::propertyValueFormatError(
+                    asyncResp->res,
+                    (*openbmc).dump(2, ' ', true,
+                                    nlohmann::json::error_handler_t::replace),
+                    "ReadyToRemove");
+                return;
+            }
+
+            if (!readytoremove)
+            {
+                BMCWEB_LOG_ERROR("Property Missing ");
+                messages::propertyMissing(asyncResp->res, "ReadyToRemove");
+                return;
+            }
+
+            // Handle special case for tod_battery assembly OEM ReadyToRemove
+            // property. NOTE: The following method for the special case of the
+            // tod_battery ReadyToRemove property only works when there is only
+            // ONE adcsensor handled by the adcsensor application.
+            if (sdbusplus::message::object_path(assembly).filename() ==
+                "tod_battery")
+            {
+                doBatteryCM(asyncResp, assembly, readytoremove.value());
+            }
+            else
+            {
+                BMCWEB_LOG_ERROR(
```

Comment:
Not an error here since user error, DEBUG here

Reply:
Fixed